src/components/PostDetails.tsx

```diff
@@ -1,117 +1,203 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Comment, CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+import { addComment, deleteComment, getComments } from '../utils/serverHelper';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+type Props = {
+  post: Post | null,
+  postSelected: Post | null,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  postSelected,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isVisibleLoader, setIsVisibleLoader] = useState(false);
+  const [isVisibleComments, setIsVisibleComments] = useState(false);
+  const [isVisibleNewForm, setIsVisibleNewForm] = useState(false);
+  const [isVisibleWriteComment, setIsVisibleWriteComment] = useState(false);
+  const [isVisibleCommentError, setIsVisibleCommentError] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isVisibleEmptyCommentMessage,
+    setIsVisibleEmptyCommentMessage] = useState(false);
+  const [isLoadingNewComment, setIsLoadingNewComment] = useState(false);
+
+  const loadComments = async (selectedPost: Post) => {
+    setIsVisibleLoader(true);
+    setIsVisibleEmptyCommentMessage(false);
+    setIsVisibleComments(false);
+
+    try {
+      const commentsFromServer = await getComments(selectedPost.id);
+
+      setComments(commentsFromServer);
+      setIsVisibleCommentError(false);
+      setIsVisibleWriteComment(true);
+      if (commentsFromServer.length === 0) {
+        setIsVisibleEmptyCommentMessage(true);
+      } else {
+        setIsVisibleComments(true);
+      }
```

Comment:
what's the main purpose of these variables? do you need them only for defining what component should be visible: component with comments or message about empty comments?

if yes, can we avoid creating 2 additional variables for this?
probably you can just check the length of the comment before `return` statement, store the result in some boolean variable and show needed component depending on those variable

Reply:
thanks for advice, I significantly reduced the amount of code